main.tf

```diff
@@ -136,6 +136,16 @@ resource "azurerm_kubernetes_cluster" "main" {
       subnet_id    = var.ingress_application_gateway_subnet_id
     }
   }
+  dynamic "key_vault_kms" {
+    for_each = var.key_vault_kms_enabled ? ["key_vault_kms"] : []
```

Comment:
Since we could turn `enabled` to `false` meanwhile keeping this `key_vault_kms`, I would recommend the following toggle expression:

```hcl
for_each = var.key_vault_kms_enabled != null ? ["key_vault_kms"] : []
```

Reply:
refactored this anyways (source was the upstream implemantation)